Classes/Form/Field/ColumnPosition.php

```diff
@@ -32,8 +32,8 @@ class ColumnPosition extends UserFunction
      */
     public function buildConfiguration()
     {
-        $fieldConfiguration = $this->prepareConfiguration('user');
-        $fieldConfiguration['userFunc'] = UserFunctions::class . '->renderColumnPositionField';
+        $fieldConfiguration = $this->prepareConfiguration('passthrough');
```

Comment:
Causes the field to not be rendered at all - type must be `user`. Since it's the same amount of code and we don't need to access the FormEngine specifics, I would prefer to keep this as part of the UserFunctions. If nothing else then to avoid registering a node type globally which will only be used in very specific cases.

Reply:
> Causes the field to not be rendered at all - type must be user. 

Rendering fine here on 8.7 - are you saying it's not rendered on 9.3?

> Since it's the same amount of code and we don't need to access the FormEngine specifics, I would prefer to keep this as part of the UserFunctions.

I converted it to a custom FormEngine field so we can set `additionalJavaScriptPost`, which is not possible with a `user` field. Of course we could just insert `<script>` tags into the rendered HTML in the user function instead, but I'm not sure how "clean" that is. Other option, as you mentioned, would be to have a global JS loaded and use `MutationObserver` or sth. similar to react to DOM changes. Also not that nice IMO but since the FormEngine js does not have any hooks to wire this up directly, it could very well be the best solution.

Reply:
> Rendering fine here on 8.7 - are you saying it's not rendered on 9.3?

Yes - when type is `passthrough` there is no HTML output on 9.3 (possibly all of 9.x series).

Reply:
> FormEngine js does not have any hooks to wire this up directly

That is exactly the problem. In `Preview` we do it like this - and I'd be fine with that same strategy in this user function:

```php
    /**
     * @return void
     */
    protected function attachAssets()
    {
        if (false === static::$assetsIncluded) {
            $doc = GeneralUtility::makeInstance(ModuleTemplate::class);
            $doc->backPath = $GLOBALS['BACK_PATH'] ?? '';

            /** @var PageRenderer $pageRenderer */
            $pageRenderer = $doc->getPageRenderer();

            $fullJsPath = PathUtility::getRelativePath(
                PATH_typo3,
                GeneralUtility::getFileAbsFileName('EXT:flux/Resources/Public/js/')
            );

            // requirejs
            $pageRenderer->addRequireJsConfiguration([
                'paths' => [
                    'FluidTypo3/Flux/FluxCollapse' => $fullJsPath . 'fluxCollapse',
                ],
            ]);
            $pageRenderer->loadRequireJsModule('FluidTypo3/Flux/FluxCollapse');

            static::$assetsIncluded = true;
        }
    }
```

Reply:
OK, then I'll try it. Do you know of an easy way to get information attached to the root form? I don't want to hard-code stuff like the maximum colPos in the JS, but have to get it from somewhere.

Reply:
I'd say we just throw all of the properties on the `<input type="hidden" />` tag as `data-` properties, so we've got `data-occupied-values`, `data-min-value`, `data-max-value` for example. The values will be the same (until parent is saved) but a tiny bit of duplication in local scope is I think preferable here, as opposed to setting these on a higher level of the rendered FormEngine stack.